### PR TITLE
simplify accelerator steps

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

simplify train/val/test steps as they are redundant

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/horovod_accelerator.py [hunk 1]
```diff
@@ -114,46 +114,28 @@ def train(self):
         hvd.join()
         return results
 
-    def training_step(self, args):
+    def _step(self, model_step: Callable, args):
         if self.trainer.on_gpu:
             batch = args[0]
             batch = self.batch_to_device(batch, hvd.local_rank())
             args[0] = batch
```
Comment (rohitgr7) [inline pytorch_lightning/accelerators/horovod_accelerator.py:121]:
```suggestion
            args[0] = self.batch_to_device(args[0], hvd.local_rank())
```

**Review by rohitgr7 — COMMENTED** [review]
GPU and TPU accelerator can be simplified a bit too.

File: pytorch_lightning/accelerators/horovod_accelerator.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/horovod_accelerator.py:121]: good batch !

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by Borda** [comment]
> GPU and TPU accelerator can be simplified a bit too.

mind send a PR? :]

**Review by rohitgr7 — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**